src/track/cueinfo.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+// cueinfo.h
+// Created 2020-02-28 by Jan Holthuis
+
+#include "util/color/color.h"
+
+// DTO for Cue information without dependencies on the actual Track object
+class CueInfo : public QObject {
+    Q_OBJECT
+
+  public:
+    enum class Type {
+        Invalid = 0,
+        HotCue = 1,
+        MainCue = 2,
+        Beat = 3, // unused (what is this for?)
+        Loop = 4,
+        Jump = 5,
+        Intro = 6,
+        Outro = 7,
+        AudibleSound = 8, // range that covers beginning and end of audible
+                          // sound; not shown to user
+    };
+
+    static constexpr double kNoPosition = -1.0;
+    static const int kNoHotCue = -1;
+
+    CueInfo();
+    CueInfo(CueInfo::Type type,
+            double startPositionMilliseconds,
+            double endPositionMilliseconds,
+            int hotCueNumber,
+            QString label,
+            PredefinedColorPointer color);
+
+    CueInfo::Type getType() const;
+    void setType(CueInfo::Type type);
+
+    void setStartPosition(double positionMilliseconds);
+    double getStartPosition() const;
+
+    void setEndPosition(double positionMilliseconds);
+    double getEndPosition() const;
+
+    int getHotCueNumber() const;
+    void setHotCueNumber(int hotCueNumber);
+
+    QString getLabel() const;
+    void setLabel(QString label);
+
+    PredefinedColorPointer getColor() const;
+    void setColor(PredefinedColorPointer color);
+
+  private:
+    Type m_type;
+    double m_startPositionMilliseconds;
+    double m_endPositionMilliseconds;
+    int m_hotCueNumber;
+    QString m_label;
+    PredefinedColorPointer m_color;
```

Comment:
I'm unsure if this dependency belongs in here. `CueInfo` should not be aware of the fact that a palette of predefined colors exists in Mixxx. It is supposed to be used in the context of external libraries where this information is not available.

I would prefer a universal `RgbColor::optional_t` here. The mapping to a predefined color should happen when converting the light-weight `CueInfo` into a heavy-weight, internal `Cue` object with all its connections and dependencies.

Reply:
Oops, you're right. We're switching to `RgbColor` anyway and I somehow came to the conclusion that we'd need to wait for #2520 to get rid of this, but that's not the case here. I'll change this.

Reply:
CueInfo should be as universal as possible without hard-coding any implicit assumptions or current internals of Mixxx. It's a path into the future.

Reply:
> It's a path into the future.

:blush: